airflow/providers/databricks/hooks/databricks.py

```diff
@@ -168,13 +174,45 @@ def _do_api_call(self, endpoint_info, json):
 
         self.databricks_conn = self.get_connection(self.databricks_conn_id)
 
+        headers = USER_AGENT_HEADER.copy()
+
         if 'token' in self.databricks_conn.extra_dejson:
             self.log.info('Using token auth. ')
             auth = _TokenAuth(self.databricks_conn.extra_dejson['token'])
             if 'host' in self.databricks_conn.extra_dejson:
                 host = self._parse_host(self.databricks_conn.extra_dejson['host'])
             else:
                 host = self.databricks_conn.host
+        elif (
+            'azure_client_id' in self.databricks_conn.extra_dejson
+            and 'azure_client_secret' in self.databricks_conn.extra_dejson
+            and 'azure_tenant_id' in self.databricks_conn.extra_dejson
+        ):
+            self.log.info('Using AAD Token for SPN. ')
+
+            if 'host' in self.databricks_conn.extra_dejson:
+                host = self._parse_host(self.databricks_conn.extra_dejson['host'])
```

Comment:
To be honest I did not understand why `host` is allowed as an extra variable for token authentication - I suspected something about backward compatibility. For user / password authentication, `host` may only be submitted as a normal connection argument. I would prefer to do it the same way with SP based authentication.

Reply:
yeah, looks like more for compatibility...